broadbean/broadbean.py

```diff
@@ -1850,6 +1958,163 @@ def outputForAWGFile(self):
                                                       DCgain=1)
                     elements[pos][chan][0] = postfilter
 
+        return elements
+
+    def outputForSEQXFile(self) -> Tuple:
+        """
+        Generate a tuple matching the call signature of the QCoDeS
+        AWG70000A driver's `makeSEQXFile` function. If channel delays
+        have been specified, they are added to the ouput before exporting.
+        The intended use of this function together with the QCoDeS driver is
+
+        .. code:: python
+
+            pkg = seq.outputForSEQXFile()
+            seqx = awg70000A.makeSEQXFile(*pkg)
+
+        Returns:
+            A tuple holding (trig_waits, nreps, event_jumps, event_jump_to,
+                go_to, wfms, amplitudes, seqname)
+        """
+
+        # most of the footwork is done by the following function
+        elements = self._prepareForOutputting()
+        # _prepareForOutputting asserts that channel amplitudes and
+        # full sequencing is specified
+        seqlen = len(elements)
+        # all elements have ident. chans since _prepareForOutputting
+        # did not raise an exception
+        channels = self.element(1).channels
+
+        for chan in channels:
+            offkey = 'channel{}_offset'.format(chan)
+            if offkey in self._awgspecs.keys():
+                log.warning("Found a specified offset for channel "
+                            "{}, but .seqx files can't contain offset "
+                            "information. Will ignore the offset."
+                            "".format(chan))
```

Comment:
Is this dangerous if the offset is discarded. Should it raise?

Reply:
I don't think it's very dangerous. And it'd be cumbersome to remove the offset, so I think a warning is sufficient.